lib/rerouter.js

```diff
@@ -0,0 +1,57 @@
+/* @flow weak */
+"use strict";
+var rerouter = module.exports = { };
+
+var router = require("./router.js");
+var jsonApi = require("./jsonApi.js");
+var url = require("qs");
+
+
+rerouter.route = function(newRequest, callback) {
+  var validRoutes = router._routes[newRequest.method.toLowerCase()];
+
+  var path = newRequest.uri;
+  if (path.match(/^https?\:\/\//)) {
+    path = path.split("/").slice(3).join("/");
+  }
+  if (jsonApi._apiConfig.base !== "/") {
+    if (path[0] !== "/") path = "/" + path;
+    path = path.split(jsonApi._apiConfig.base);
+    path.shift();
+    path = path.join(jsonApi._apiConfig.base);
+  }
+  path = path.replace(/^\//, "").split("?")[0].replace(/\/$/, "");
```

Comment:
All this string manipulation is unfortunate, but hard to avoid. The internal strings were designed to make it really easy to glue them together to generate user-facing URLs, not destructure user-facing URLs. Oh well.